### DISCUSSION TITLE
Hardware acceleration using Plex HA add-on

### DISCUSSION THREAD

**Discussion by agf102** [hassio-addons/addon-plex#200] [Q&A] [answered]
Hi,

I've read the documentation and searched the internet but I can't find whether the Plex HA add-on has access to the CPU HW encode / decode acceleration.  Anyone know?

**Comment by frenck** [comment, marked as answer]
Sorry, it doesn't have access to that.

**Comment by zsolth** [comment]
Hi @frenck any more details on why not? It seems like during startup plex does find the device

```
DEBUG - [GPU] Got device: HD Graphics 630, intel@builtin, default true, best true, ID 8086:5912:1028:07a3@0000:00:02.0, DevID [8086:5912:1028:07a3], flags 0x38e7
INFO - Preemptively preparing driver imd for GPU HD Graphics 630
DEBUG - [DriverDL/imd] Skipping download; already exists
INFO - Preemptively preparing driver icr for GPU HD Graphics 630
DEBUG - [DriverDL/icr] Skipping download; already exists
```


but then during playback fails with:

```
DEBUG - [Req#c70/Transcode] Codecs: testing h264 (decoder) with hwdevice vaapi
DEBUG - [Req#c70/Transcode] Codecs: hardware transcoding: testing API vaapi for device '/dev/dri/renderD128' (HD Graphics 630)
ERROR - [Req#c70/Transcode] [FFMPEG] - No VA display found for device /dev/dri/renderD128.
DEBUG - [Req#c70/Transcode] Codecs: hardware transcoding: opening hw device failed - probably not supported by this system, error: Invalid argument
DEBUG - [Req#c70/Transcode] Could not create hardware context for h264
```

In fact the driver is there in /dev
```
# pwd
/dev/dri
# ls -l
total 0
drwxr-xr-x    2 root     root            80 Nov  6 17:55 by-path
crw-rw----    1 root     video     226,   0 Nov  6 17:55 card0
crw-rw-rw-    1 root     render    226, 128 Nov  6 17:55 renderD128
```


I have the latest available version of the addon, HAOS and Core

Core: 2023.11.1
Supervisor: 2023.10.1
Addon: 3.3.2

I'd be happy to help debug / test this if that helps.

Thanks,
Zsolt

**Reply by diamant-x** [reply]
From a container perspective, such feature seems to be supported officially by plex https://github.com/plexinc/pms-docker#intel-quick-sync-hardware-transcoding-support. However they do mention:

> Intel Quick Sync support also requires newer 64-bit versions of the Ubuntu or Fedora Linux operating system to make use of this feature.

`So I wonder if maybe HAOS is not such an OS and hence can't passthrough the iGPU codecs or stuff needed for it to work in the container?

I would also be interested on seeing this happening. Although not sure why it seems plex is not that much used in HAOS https://analytics.home-assistant.io/add-ons/ I wonder if it's because HAOS not supporting HW transcoding, then people just install it in another system that has it. But from systems perspective it definitely would have potential value for nearly 50% of the installed base which doesn't run on ARM-based device as per https://analytics.home-assistant.io/

**Comment by cannondale0815** [comment]
I'll jump in here to just add my "+1" to the mix. It would be outstanding if there was a way, even if complex, to enable the Plex HA Add-On to work with the Intel iGPU (QuickSyncView"), if the host is equipped with such a CPU.

EDIT: Came back here to say that hardware transcoding in the Plex Add-On for HA is, in fact, working. I have so far only tried on LAN where I forced the web player to transcode to a lower format, but it does show that video is being hardware transcoded.

**Reply by chrisdmorris-wq** [reply]
Did you get this working correctly? I'm showing HW transcode but it buffers every few seconds. Neither Plex nor Home Assistant are showing any real load on the hardware or network.

**Reply by cannondale0815** [reply]
Yes, HW transcoding is working. If you see buffering, it may have to do with network bottlenecks or WiFi congestion where your player is.
